macOS/dyld_cache_map_parser.cpp

```diff
@@ -34,6 +34,35 @@ static bool is_page_aligned(uint64_t addr, uint64_t page_size = 0x4000) {
   return !(addr & (page_size-1));
 }
 
+std::string find_dyld_map() {
+  std::string name = "/System/Library/dyld/dyld_shared_cache_arm64e.map";
+
+  // from dyld4::APIs::dyld_shared_cache_find_iterate_text
+  std::string cryptexPrefixes[] = {
+    "/System/Volumes/Preboot/Cryptexes/OS/",
+    "/private/preboot/Cryptexes/OS/",
+    "/System/Cryptexes/OS"
+  };
+
+  auto check = [](const std::string &path) {
+    std::ifstream stream(path);
+    return !stream.fail();
+  };
+
+  if (check(name)) {
+    return name;
+  }
+
+  for (auto i = 0; i < 3; i++) {
+    std::string cryptex = cryptexPrefixes[i] + "/" + name;
```

Comment:
Follow-up from the previous comment, `name` already starts with a slash, so it doesn't need to be appended here, e.g. if you take
`"/System/Volumes/Preboot/Cryptexes/OS/" + "/" + "/System/Library/dyld/dyld_shared_cache_arm64e.map"` you end up with three consecutive slashes,
`/System/Volumes/Preboot/Cryptexes/OS///System/Library/dyld/dyld_shared_cache_arm64e.map`

Reply:
Did a force push to remove redundant slashes. Thanks!